lib/ProductOpener/Food.pm

```diff
@@ -1136,69 +1193,107 @@ Ref to a mapping suitable to call compute_nutriscore_score_and_grade
 
 =cut
 
-sub compute_nutriscore_data ($product_ref, $prepared, $nutriments_field) {
+sub compute_nutriscore_data ($product_ref, $prepared, $nutriments_field, $version = "2021") {
 
 	my $nutriments_ref = $product_ref->{$nutriments_field};
 
-	# compute data
-	my $saturated_fat_ratio = saturated_fat_ratio($nutriments_ref, $prepared);
-	my $fruits = compute_fruit_ratio($product_ref, $prepared);
-	my $nutriscore_data = {
-		is_beverage => $product_ref->{nutrition_score_beverage},
-		is_water => is_water_for_nutrition_score($product_ref),
-		is_cheese => is_cheese_for_nutrition_score($product_ref),
-		is_fat => is_fat_for_nutrition_score($product_ref),
-
-		energy => $nutriments_ref->{"energy" . $prepared . "_100g"},
-		sugars => $nutriments_ref->{"sugars" . $prepared . "_100g"},
-		saturated_fat => $nutriments_ref->{"saturated-fat" . $prepared . "_100g"},
-		saturated_fat_ratio => $saturated_fat_ratio,
-		sodium => $nutriments_ref->{"sodium" . $prepared . "_100g"} * 1000,    # in mg,
-
-		fruits_vegetables_nuts_colza_walnut_olive_oils => $fruits,
-		fiber => (
-			(defined $nutriments_ref->{"fiber" . $prepared . "_100g"})
-			? $nutriments_ref->{"fiber" . $prepared . "_100g"}
-			: 0
-		),
-		proteins => $nutriments_ref->{"proteins" . $prepared . "_100g"},
-	};
+	my $nutriscore_data_ref;
+
+	# The 2021 and 2023 version of the Nutri-Score need different nutrients
+	if ($version eq "2021") {
+		# fruits, vegetables, nuts, olive / rapeseed / walnut oils
+		my $fruits = compute_fruit_ratio($product_ref, $prepared);
+
+		my $is_fat = is_fat_for_nutrition_score($product_ref);
+
+		$nutriscore_data_ref = {
+			is_beverage => $product_ref->{nutrition_score_beverage},
+			is_water => is_water_for_nutrition_score($product_ref),
+			is_cheese => is_cheese_for_nutrition_score($product_ref),
+			is_fat => $is_fat,
+
+			energy => $nutriments_ref->{"energy" . $prepared . "_100g"},
+			sugars => $nutriments_ref->{"sugars" . $prepared . "_100g"},
+			saturated_fat => $nutriments_ref->{"saturated-fat" . $prepared . "_100g"},
+			sodium => $nutriments_ref->{"sodium" . $prepared . "_100g"} * 1000,    # in mg,
+
+			fruits_vegetables_nuts_colza_walnut_olive_oils => $fruits,
+			fiber => (
+				(defined $nutriments_ref->{"fiber" . $prepared . "_100g"})
+				? $nutriments_ref->{"fiber" . $prepared . "_100g"}
+				: 0
+			),
+			proteins => $nutriments_ref->{"proteins" . $prepared . "_100g"},
+		};
```

Comment:
why not do initialization of $nutriscore_data_ref outside of the `if ($version eq "2021")` with all the thing that are common to both versions ?

Reply:
We won't keep both versions in the long run, so I thought it would be easier to keep things clearly separated, even if there are many common nutrients.